repository-basex/src/test/java/org/lfenergy/compas/scl/data/repository/CompasSclDataBaseXRepositoryTest.java

```diff
@@ -0,0 +1,234 @@
+// SPDX-FileCopyrightText: 2021 Alliander N.V.
+//
+// SPDX-License-Identifier: Apache-2.0
+package org.lfenergy.compas.scl.data.repository;
+
+import org.basex.BaseXServer;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.lfenergy.compas.commons.MarshallerWrapper;
+import org.lfenergy.compas.scl.SCL;
+import org.lfenergy.compas.scl.data.basex.BaseXClientFactory;
+import org.lfenergy.compas.scl.data.model.ChangeSetType;
+import org.lfenergy.compas.scl.data.model.SclType;
+import org.lfenergy.compas.scl.data.model.Version;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.lfenergy.compas.scl.data.basex.BaseXServerUtil.createServer;
+import static org.lfenergy.compas.scl.data.basex.BaseXServerUtil.getFreePortNumber;
+
+@ExtendWith(MockitoExtension.class)
+class CompasSclDataBaseXRepositoryTest {
+    private static final SclType TYPE = SclType.SCD;
+    private static BaseXServer server;
+    private static BaseXClientFactory factory;
+    private CompasSclDataBaseXRepository repository;
+
+    @BeforeAll
+    static void beforeAll() throws Exception {
+        var portNumber = getFreePortNumber();
+        server = createServer(portNumber);
+        factory = new BaseXClientFactory("localhost", portNumber, "admin", "admin");
```

Comment:
No a fan of doing thus kind of stuff in a abstract class, because we can only extend once.
Better to do that in Utility classes.
Also move creating the client factory to the Utility class. This way it will be 3 lines of calling methods.